numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -368,6 +368,9 @@ array_implement_array_function(
              */
             public_api = implementation;
         }
+        if (like_arg == Py_None){
+            PyDict_DelItem(kwargs, npy_ma_str_like);
+        }
```

Comment:
Hmmm, coverage does not hit this code.  I suppose because the Python code will never call the dispatching machinery with `like=None` anyway.

Reply:
hmm yeah I guess this could be removed then if you agree @seberg ?

Reply:
Seems OK either way.  Could also add an assert (if it makes things clearer for later).